make/mt-os/mt-osx.mk

```diff
@@ -33,4 +33,6 @@ endif
 
 IOTC_CONFIG_FLAGS += -DIOTC_MULTI_LEVEL_DIRECTORY_STRUCTURE
 IOTC_CONFIG_FLAGS += -DIOTC_LIBCRYPTO_AVAILABLE
+IOTC_CONFIG_FLAGS += -DWOLFSSL_X86_64_BUILD
```

Comment:
here as well please: 

```
ifneq (,$(findstring wolfssl,$(IOTC_TLS_LIB_NAME)))
  # need for x86 wolfssl builds
  IOTC_CONFIG_FLAGS += -DWOLFSSL_X86_64_BUILD
endif

Also, should this section ^ also be added to mt-freertos-linux.mk in this directory as well? 
```

Reply:
Since `freertos` can run on a number of non-x86 based architectures (where-as most, if not all, of our Linux/OSX targets would be x86 based), we would need an architecture specific build flag as well. I am not familiar enough with that system to surmise as much, but I can add it if you'd like as it wouldn't be worse off than it was before.

Reply:
Added the guards. I opted to use the same logical check for which crypto library is used from `[mt-crypto](https://github.com/GoogleCloudPlatform/iot-device-sdk-embedded-c/blob/master/make/mt-config/mt-crypto.mk#L18)`

Reply:
Makes sense. Added.